test/e2e/scheduler_predicates.go

```diff
@@ -219,24 +219,29 @@ var _ = framework.KubeDescribe("SchedulerPredicates [Serial]", func() {
 
 		By(fmt.Sprintf("Starting additional %v Pods to fully saturate the cluster max pods and trying to start another one", podsNeededForSaturation))
 
-		framework.StartPods(c, podsNeededForSaturation, ns, "maxp", api.Pod{
-			TypeMeta: unversioned.TypeMeta{
-				Kind: "Pod",
-			},
-			ObjectMeta: api.ObjectMeta{
-				Name:   "",
-				Labels: map[string]string{"name": ""},
-			},
-			Spec: api.PodSpec{
-				Containers: []api.Container{
-					{
-						Name:  "",
-						Image: "gcr.io/google_containers/pause:2.0",
+		// As the pods are distributed randomly among nodes,
+		// it can easily happen that all nodes are satured
+		// and there is no need to create additional pods.
+		// StartPods requires at least one pod to replicate.
+		if podsNeededForSaturation > 0 {
```

Comment:
if I'm understanding this PR correctly, all that's been done is to effectively move the conditional from `StartPods` out to the callers. what benefit does this yield?

put another way: before this PR, `StartPods` just returns if `replicas` is 0. with this PR, we don't call `StartPods` if `replicas` is zero. but the behavior is effectively the same, so why do we need to do this?